jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -377,7 +378,7 @@ def add_data(self, data, color=None, alpha=None, **layer_state):
         # that new data entries (from model fitting or gaussian smooth, etc) will only be spectra
         # and all subsets affected will be spectral
         for layer in self.state.layers:
-            if "Subset" in layer.layer.label and layer.layer.data.label == data.label:
+            if get_subset_type(layer.layer) == 'spectral' and layer.layer.data.label == data.label:
```

Comment:
I think `get_subset_type` assumes the input is a subset... does the fact that this works depend on the assumption that spectral is defined first in that function (do we need some way to still detect if the layer is a subset at all vs actual data)?

Reply:
`get_subset_type` has a check to start if the `layer.layer` object has attribute "subset_state" and if not it returns None. I added a check to this if statement though to make it more clear that this should only accept layer's associated with subsets.